src/org/pahappa/systems/registrationapp/views/UserView.java

```diff
@@ -68,19 +72,45 @@ public void displayMenu() {
         }
     }
 
+    /** helper function for handling user retries */
+    private boolean retry() {
+        System.out.println("An error has been encountered with your inputs. Please hit r to try again.");
+        String retry = scanner.nextLine().strip();
+        return retry.toLowerCase().equals("r");
+    }
+
+    /** Removes all spaces from the input string */
+    private String spacesCleaner(String input) {
+        return input
+        .replaceAll(" ","") //space
+        .replaceAll(" ",""); // Alt+0160 chracter, looks like space
+    }
```

Comment:
The second replaceAll is redundant, 
private String spacesCleaner(String input) {
    // Use a regular expression to replace both space (U+0020) and non-breaking space (U+00A0)
    return input.replaceAll("[\u0020\u00A0]", "");
}
this would be better if you wanted to remove space and non breaking space